ios/brave-ios/Sources/Brave/Frontend/Browser/HomePanel/NTPDataSource.swift

```diff
@@ -104,114 +111,112 @@ public class NTPDataSource {
   // This can 'easily' be adjusted to support both sets by switching to String, and using filePath to identify uniqueness.
   private var lastBackgroundChoices = [Int]()
 
-  private enum ImageRotationStrategy {
-    /// Special strategy for sponsored images, uses in-memory property to keep track of which image to show.
-    case sponsoredRotation
-    /// Uses random images, keeps in-memory list of recently viewed images to avoid showing it too often.
-    case randomOrderAvoidDuplicates
+  func shouldAttemptSponsoredMedia() -> Bool {
+    return
+      Preferences.NewTabPage.backgroundMediaType.isSponsored
+      && Preferences.NewTabPage.backgroundRotationCounter.value
+        == service.initialCountToBrandedWallpaper
   }
 
-  func newBackground() -> NTPWallpaper? {
-    if !Preferences.NewTabPage.backgroundImages.value { return nil }
+  func getSponsoredMediaBackground() -> NTPWallpaper? {
+    guard let sponsor = service.sponsoredImageData,
```

Comment:
nit: sponsoredImageData instead of sponsor

Reply:
Fixed